guarddog/analyzer/metadata/npm/empty_information.py

```diff
@@ -0,0 +1,20 @@
+""" Empty Information Detector
+
+Detects if a package contains an empty description
+"""
+import os.path
+from typing import Optional
+
+from guarddog.analyzer.metadata.empty_information import EmptyInfoDetector
+
+MESSAGE = "This package has an empty description on PyPi"
+
+
+class NPMEmptyInfoDetector(EmptyInfoDetector):
```

Comment:
How about having a common baseclass for NPM packages instead? I feel like we should have `Detector -> NpmDetector -> EmptyInfoDetector` instead of `Detector -> EmptyInfoDetector -> NpmEmptyInfoDetector`

Otherwise we're mangling business logic at several inheritance layers, and assuming that every heuristic makes sense to implement both for NPM and PyPI (which is likely to be an incorrect assumption)

Reply:
One argument in favor of the current implementation might be "we reuse code for common heuristics", but I don't think the level of code reuse is worth it here

Reply:
For this one it's true. but for instance for the email domain one, we reuse a lot. So I'm still more inclined for the `Detector -> EmptyInfoDetector -> NpmEmptyInfoDetector`. I can try to use multiple inheritence here however https://www.programiz.com/python-programming/multiple-inheritance

Reply:
Then probably the logic should be extracted in a function instead of a common base class?